### ISSUE TITLE
Training loop from scratch for multi-out model - how to calculate loss and update it. 

### ISSUE THREAD

**Issue by Suzan009** [keras-team/keras#14226]
I have a ML models which have 3 output. Each of them have passed into a Soft-Max player. I'm trying to train the model from scratch with training loop in tf.keras. My generator output are something like this:

```
x,y = (batch_size, h,w,c), [(batch_size, 2), (batch_size, 8), (batch_size, 10)]
```

As you notice, each of the 3 output has different numbers of labels. First output layer has 2 labels output, second one has 8 and last one has 10. The problem is not about the data generator; It's fine. **However, I don't understand how will I calculate loss function and update it in scratch training.**

---

I've started with [document guidelines](https://keras.io/guides/writing_a_training_loop_from_scratch/). Here is the code snippet,

```python

epochs = 1
optimizer   = tf.keras.optimizers.Adam(learning_rate=0.05)
loss_fn     = tf.keras.losses.CategoricalCrossentropy()

train_acc_metric = tf.keras.metrics.Accuracy()
val_acc_metric   = tf.keras.metrics.Accuracy()

@tf.function
def train_step(x, y):
    with tf.GradientTape() as tape:
        logits = model(x, training=True)  # Logits for this minibatch

        # Compute the loss value for this minibatch.
        train_loss_value = loss_fn(y, logits)

    grads = tape.gradient(train_loss_value, model.trainable_weights)
    optimizer.apply_gradients(zip(grads, model.trainable_weights))
    train_acc_metric.update_state(y, logits)
    return train_loss_value
```

And, 

```python

for epoch in range(epochs):
    # Iterate over the batches of the dataset.
    for step, (x_batch_train, y_batch_train) in enumerate(train_generator):
        train_loss_value = train_step(x_batch_train, y_batch_train)

    # Reset metrics at the end of each epoch
    train_acc_metric.reset_states()
```

I suspect, I am not handling properly the computation of loss in here: `loss_fn(y, logits)`. `y` has 3 output, `logits` has also 3 output. 

Using `model.compile` is simply straight-forward, just passing the loss function in the loss argument. But what to do in scratch training. The 3 output (though are co-related) needs to trained separately. 

Asked [SO](https://stackoverflow.com/q/64065473/9215780), nobody knows.

**Comment by Suzan009** [comment]
The `logits = model(x, training=True)` has 3 output. So to calculate the loss function separately (I think) we need to unpack it. `loss_one, loss_two, loss_thr = model(x)`. And next, maybe like this:

```
train_loss_value_one = loss_fn(y[0], loss_one)
train_loss_value_two = loss_fn(y[1], loss_two)
train_loss_value_thr = loss_fn(y[2], loss_thr)
```
I'm feeling I'm definitively missing something.

**Comment by Suzan009** [comment]
@fchollet
